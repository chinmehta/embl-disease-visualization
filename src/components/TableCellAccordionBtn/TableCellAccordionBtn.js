import React from "react";

function TableCellAccordionBtn({ text, clickHandler }) {
  return (
    <>
      <td className="flex items-stretch border border-gray-400 border-collapse p-0">
        <button
          className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
          onClick={() => clickHandler()}
        >
          {text}
        </button>
      </td>
    </>
  );
}

export default TableCellAccordionBtn;
