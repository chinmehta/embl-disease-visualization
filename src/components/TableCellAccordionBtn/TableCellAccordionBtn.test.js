import React from "react";
import renderer from "react-test-renderer";

import TableCellAccordionBtn from "./TableCellAccordionBtn";

it("renders when no text is passed as prop", () => {
  const element = renderer.create(<TableCellAccordionBtn />).toJSON();

  expect(element).toMatchSnapshot();
});

it("renders when text and row id is passed as prop", () => {
  const element = renderer.create(<TableCellAccordionBtn text="+" />).toJSON();

  expect(element).toMatchSnapshot();
});
