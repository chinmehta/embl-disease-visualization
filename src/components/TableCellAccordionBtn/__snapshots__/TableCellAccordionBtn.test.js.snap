// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when no text is passed as prop 1`] = `
<td
  className="flex items-stretch border border-gray-400 border-collapse p-0"
>
  <button
    className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
    onClick={[Function]}
  />
</td>
`;

exports[`renders when text and row id is passed as prop 1`] = `
<td
  className="flex items-stretch border border-gray-400 border-collapse p-0"
>
  <button
    className="flex text-xl px-3 py-3 h-100 w-100 bg-embl-primary-blue text-white w-full h-full justify-center "
    onClick={[Function]}
  >
    +
  </button>
</td>
`;
